man/fifelse.Rd

```diff
@@ -37,6 +37,39 @@ data.table::fifelse(v \%in\% c("a","b","c"), v, factor(NA))
 # Levels: a b c d e
 
 \dontrun{
+# Example 3:
+# Unlike dplyr::if_else and hutils::if_else, data.table::fifelse and base::ifelse
+# allow singleton replacements to be en-listed ( i.e. wrapped correctly in list()).
+
+base::ifelse(c(TRUE, FALSE), 1L, list(0L))
+# [[1]]
+# [1] 1
+# 
+# [[2]]
+# [1] 0
+
+data.table::fifelse(c(TRUE, FALSE), 1L, list(0L))
+# [[1]]
+# [1] 1
+# 
+# [[2]]
+# [1] 0
+
+dplyr::if_else(c(TRUE, FALSE), 1L, list(0L))
+# Error: `false` must be an integer vector, not a list
+# Call `rlang::last_error()` to see a backtrace
+
+hutils::if_else(c(TRUE, FALSE), 1L, list(0L))
+# Error in hutils::if_else(c(TRUE, FALSE), 1L, list(0L)) : 
+#   typeof(false) == list, yet typeof(true) == integer. Both true and false must have the same type.
+
+# The above is useful given that data.table allows list columns as shown below:
+DT <- data.table::data.table(a = 0:5, b = list(1, list(2:3), list(4:6), list(6:4), list(3:2), 1))
+all.equal(DT[ , b := data.table::fifelse(a > 2, b, 0)],DT[ , b := base::ifelse(a > 2, b, 0)])
```

Comment:
I guess this all.equal test does not work. The code for the b column could give very different results, but we would still see TRUE because b is overwritten in-place. Eg, `DT = data.table(x = 1); all.equal(DT[, b := 1], DT[, b := "yo"])`.

Reply:
Thank you. I corrected it. Hope it is better now!